src/mdsynthesis/tests/test_treants.py

```diff
@@ -353,8 +319,7 @@ def test_fresh_sim_readonly(self, sim):
 
         assert isinstance(s.universe, mda.Universe)
 
-        # since we didn't add any atom selections, should raise KeyError
-        # would be nice if it DIDN'T behave this way, but lazy loading keeps
-        # Sim init cheaper
-        with pytest.raises(KeyError):
-            s.atomselections
+    @pytest.mark.skip("impl in datreant.core")
+    def test_write_as_readonly(self, sim):
+        with pytest.raises(IOError):
+            sim.atomselections['foo'] = 'bar'
```

Comment:
Here I change the existing behavior. In my opinion, just reading from an `AtomSelections` object should be ok even in a read-only SIM. An error should only be thrown when we want to write a new selection. This also means that if the statefile wasn't written we return a default atomselection. I think this behavior is universal to any metadata and would be best implemented in the core library. @dotsdl  what do you think?

Reply:
@kain88-de I agree with this change. Not exactly sure where to put the behavior you describe in the core library, but being able to handle missing statefiles gracefully and predictably is something we generally prefer.

Reply:
Do we have tests for readonly in the core library?

Reply:
@kain88-de we do [here](https://github.com/datreant/datreant.core/blob/develop/src/datreant/core/tests/test_treants.py#L493). I think it's good to leave this test in, so I'm going to remove the skip decorator.

Reply:
Does this test even work right now?

Reply:
Ah...not on Python 2.7. This yields an ``OSError`` instead of an ``IOError`` there. Accept either?

Reply:
yeah it passes. I'm surprised

Reply:
I think I would like in addition to have a test like this.

```python
    def test_read_as_readonly(self, sim):
        assert sim.atomselections['foo'] is None

```

The current error message isn't so informative. I get a PermissionError as expected. But I'm not 100% set that returning `None` would be better. I also note that this issue is not a blocker for me to merge this.